pkg/build/registry/buildconfig/webhook.go

```diff
@@ -72,9 +73,18 @@ func (w *WebHook) ServeHTTP(writer http.ResponseWriter, req *http.Request, ctx k
 		Revision:    revision,
 		Env:         envvars,
 	}
-	if _, err := w.instantiator.Instantiate(config.Namespace, request); err != nil {
+	newBuild, err := w.instantiator.Instantiate(config.Namespace, request)
+	if err != nil {
 		return errors.NewInternalError(fmt.Errorf("could not generate a build: %v", err))
 	}
+
+	// Send back the build name so that the client can alert the user.
+	newBuildEncoded, err := runtime.Encode(kapi.Codecs.LegacyCodec(buildapi.SchemeGroupVersion), newBuild)
```

Comment:
The version specified here looks wrong.  You need to use the version under which you were requested so that a hook under `/oapi/v1` returns `Build.v1` and `/oapi/v2` return `Build.v2`.